docs/rfd.md

```diff
@@ -0,0 +1,478 @@
+  //
+  // If 'follow' is set to true, the output will continue to be be streamed in
+  // real-time until the job completes. Otherwise, only the currently accumulated
+  // output will be written, and the stream will be closed even if the job is
+  // still running.
+  //
+  // If the job is already completed, the full output of the job will be
+  // written to the stream, after which the stream will be closed.
+  rpc Output(OutputRequest) returns (stream ProcessOutput);
+}
+
+// JobSpec describes a command to be run, along with optional resource limits
+// that should be applied to the command's process.
+message JobSpec {
+  CommandSpec    command = 1;
+  ResourceLimits limits  = 2;
+}
+
+message JobId {
+  string id = 1;
+}
+
+message JobIdList {
+  repeated JobId items = 1;
+}
+
+enum State {
+  Unknown = 0;
+  // The job is waiting to be started, and is not yet running.
+  Pending = 1;
+  // The job is currently running.
+  Running = 2;
+  // The job exited normally (with any exit code), or was terminated via signal.
+  Completed = 3;
+  // The job failed to start, or exited abnormally due to an I/O error.
+  Failed = 5;
+}
+
+message JobStatus {
+  // The job's logical state.
+  State state = 1;
+  // A human-readable message describing the job's state, or an error message
+  // if the job failed.
+  string message = 2;
+  // The PID of the job's process.
+  int32 pid = 3;
+  // The exit code of the job's process.
+  // Only present if the job is in the Completed state.
+  optional int32 exitCode = 4;
+  // A bitmask equivalent to syscall.WaitStatus, also described in wait(2).
+  // Only present if the job is in the Completed or Failed state.
+  optional uint32 waitStatus = 5;
+  // The time at which the job was started.
+  google.protobuf.Timestamp startTime = 6;
+  // The time at which the job completed (or failed). If the job is still
+  // running, this field will not be present.
+  google.protobuf.Timestamp endTime = 7;
+  // The job's original spec.
+  JobSpec spec = 8;
+}
+
+message CommandSpec {
+  // The command name to run. Required.
+  string command = 1;
+  // The command's arguments, not containing the command name itself.
+  repeated string args = 2;
+  // Optional additional environment variables to set for the command.
+  // These will be merged with the job server's environment variables.
+  repeated string env = 3;
+  // The working directory for the command. If not specified, the job server's
+  // working directory will be used.
+  optional string cwd = 4;
+  // Optional user id to run the command as.
+  // If not specified, the job server's user id will be used.
+  optional int32 uid = 5;
+  // Optional group id to run the command as.
+  // If not specified, the job server's group id will be used.
+  optional int32 gid = 6;
+  // Optional data to be written to the process's stdin pipe when it starts.
+  // If not specified, the process's stdin will be set to /dev/null.
+  optional bytes stdin = 7;
```

Comment:
Let omit these from the scope. Command and arguments are enough for this challenge 👍